### ISSUE TITLE
CA1508 false positive on String.IsNullOrWhiteSpace (in 5.0.2)

### ISSUE THREAD

**Issue by airbreather** [dotnet/roslyn-analyzers#4755]
I see that #4439 recently fixed something that sounded like this, and apparently the tag lists that it's fixed in 5.0.1.  However, my `dotnet --info` reports that I'm on version 5.0.2 (`<AnalysisLevel>latest</AnalysisLevel>`), and I still seem to be able to repro a very similar CA1508 false positive.

## Steps
1. Save the three following files to a folder.
1. `dotnet build --no-incremental`

Actual: CA1508 on the `string.IsNullOrWhiteSpace` call.
Expected: no CA1508 warnings

## My `dotnet --info` Output (not one of the three files)
```
.NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.102\

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  5.0.101 [C:\Program Files\dotnet\sdk]
  5.0.102 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

----

Here are the three files:

## ConsoleApp0.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisLevel>latest</AnalysisLevel>
  </PropertyGroup>

</Project>
```

## .editorconfig
```editor-config
[*.cs]
dotnet_diagnostic.CA1508.severity = warning
```

## Program.cs
```csharp
using System;
using System.Linq;

static class Helpers
{
    static void Main()
    {
        Helper("a");
    }

    static void Helper(string s)
    {
        if (s is null)
        {
            throw new ArgumentNullException(nameof(s));
        }

        if (string.IsNullOrWhiteSpace(s))
        {
            throw new ArgumentException("Cannot be blank", nameof(s));
        }

        Console.WriteLine(s);
    }
}
```

**Comment by airbreather** [comment]
WHOOPS.  User error.  Something smelled fishy, so I checked into this a bit more to see what was going on... the documentation on https://docs.microsoft.com/en-us/visualstudio/code-quality/install-net-analyzers?view=vs-2019 misled me a bit.

The documentation on https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#latest-updates points out that `<AnalysisLevel>latest</AnalysisLevel>` actually uses the latest analyzers **bundled with the SDK**.

Installing the latest package from NuGet actually does get me the bug fix.  Sorry for the spam.

**Comment by loraderon** [comment]
I noticed that I had to use `<AnalysisLevel>5.0.3</AnalysisLevel>` since it fails with both `<AnalysisLevel>5.0</AnalysisLevel>` and `<AnalysisLevel>latest</AnalysisLevel>`
